### PR TITLE
artemis: tests added on trip add chaining with some options

### PR DESCRIPTION
**Pull request by kadhikari**
Tests on trip add with multiple actions:

1. Add a new trip, update with a delay and update to normal
2. Add a new trip, update with a delay, update with a stop_time deleted and update to normal  
3. Add a new trip, update with a delay, update with a new stop_time and delete de trip

For details see: https://jira.kisio.org/browse/NAVP-1068

### CODE REVIEW

File: artemis/tests/fixtures/trip_add_paris_marseille.json [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+        "timeZoneLocale": "Europe/Paris",
+        "horaireVoyageurDepart": {
+          "dateHeure": "2012-11-20T11:00:00+0000",
+          "heureLocale": "12:00:00",
+          "idFuseauHoraire": "Europe/Paris",
+          "nbJoursPasseMinuit": 0,
+          "statutCirculationOPE": "CREATION"
+        },
+        "listeHoraireProjeteArrivee": [],
+        "listeHoraireProjeteDepart": [
```
Comment (pbougue) [inline artemis/tests/fixtures/trip_add_paris_marseille.json]: **#QA**: Do we always have those `horaireProjete` in COTS trip-add feeds?
It would be nice to have some stops without those to check the behavior si OK (kirin should be good I think).

File: artemis/tests/guichet_unique_test.py [hunk 2]
```diff
@@ -1052,6 +1052,260 @@ def test_kirin_cots_sequence_08(self):
...
+                     max_nb_transfers="0",
+                     data_freshness="realtime")
+
+    @xfail(reason="Waiting for NAVP-1035", raises=AssertionError, strict=True)
+    def test_kirin_cots_add_trip_chain_type_3(self):
+        """
+        1. A simple trip add with 5 stop_times all existing in navitia
+        2. Trip modified with 15 minutes delay in each stop_times
+        3. Trip modified with a new stop_time (gare de Orleans) added in the above flux cots
+        4. Delete the trip with "statutCirculationOPE": "SUPPRESSION" in all stop_times
```
Comment (pbougue) [inline artemis/tests/guichet_unique_test.py]: Same remark as on kirin: is it possible to add it back at the end, to check it's possible?
Maybe it should be done in another PR.

File: artemis/tests/fixtures/trip_add_paris_marseille_delete_with_delay_15_and_stop_time_added.json [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+        "listeMotifDepart": [
+          {
+            "dateHeureDeclaration": "2012-11-20T08:35:06+0000",
+            "idMotifInterne": 78,
+            "source": "IRE-GOP"
+          }
+        ],
+        "rang": 2,
+        "typeArret": "CH",
+        "horaireVoyageurDepart": {
```
Comment (pbougue) [inline artemis/tests/fixtures/trip_add_paris_marseille_delete_with_delay_15_and_stop_time_added.json]: minor: `horaireVoyageur` on top of every stop is easier to review.
This is the case on all this feed and on some other feeds of this PR: if you decide to address it, please check all :)

Comment (kadhikari) [reply artemis/tests/fixtures/trip_add_paris_marseille_delete_with_delay_15_and_stop_time_added.json]: It's a online app i used to format and sort objects in json

File: artemis/tests/guichet_unique_test.py (hunk 2 above)
Comment (kadhikari) [reply artemis/tests/guichet_unique_test.py]: Done

File: artemis/tests/fixtures/trip_add_paris_marseille.json (hunk 1 above)
Comment (kadhikari) [reply artemis/tests/fixtures/trip_add_paris_marseille.json]: Flux cots "trip_add_paris_marseille_to_normal.json" contains stops without "pronosticIV" and kirin is fine with that.

File: artemis/tests/fixtures/trip_add_paris_marseille_with_delay_15.json [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+        }
+      },
+      {
+        "@id": "0f4666b5-082f-4006-9e06-65702e45fdda",
+        "cr": "0087",
+        "ci": "751008",
+        "ch": "BV",
+        "timeZoneLocale": "Europe/Paris",
+        "horaireVoyageurArrivee": {
+          "dateHeure": "2012-11-20T16:00:00+0000",
```
[the remark this answers is not in the document]
Comment (kadhikari) [reply artemis/tests/fixtures/trip_add_paris_marseille_with_delay_15.json]: If really necessary, could be done in future PR -;

File: artemis/tests/guichet_unique_test.py [hunk 5]
```diff
@@ -1052,6 +1052,281 @@ def test_kirin_cots_sequence_08(self):
                      max_nb_transfers="0",
                      data_freshness="realtime")
 
+    @xfail(reason="Waiting for NAVP-1035", raises=AssertionError, strict=True)
+    def test_kirin_cots_add_trip_chain_type_1(self):
+        """
+        Test for case 11 from the file "Enchainement Cas_API_20181010.xlsx"
```
Comment (mbounabi) [inline artemis/tests/guichet_unique_test.py]: Minor: not necessary to mention the excel file name...
Also, every other tests from this file contain the case number in the test name (ex:test_kirin_cots_sequence_09).
It might not be enough to understand what the test is about, but maybe "test_kirin_cots_add_trip_sequence_11"

File: artemis/tests/guichet_unique_test.py [hunk 6]
```diff
@@ -1052,6 +1052,281 @@ def test_kirin_cots_sequence_08(self):
...
+        1. A simple trip add with 5 stop_times all existing in navitia
+        2. Trip modified with 15 minutes delay in each stop_times
+        3. Return to normal
+        """
+        """
+        Base and realtime request
```
Comment (mbounabi) [inline artemis/tests/guichet_unique_test.py]: Minor: comment line not really necessary

File: artemis/tests/guichet_unique_test.py [hunk 7]
```diff
@@ -1052,6 +1052,281 @@ def test_kirin_cots_sequence_08(self):
                      max_nb_transfers="0",
                      data_freshness="realtime")
 
+    @xfail(reason="Waiting for NAVP-1035", raises=AssertionError, strict=True)
+    def test_kirin_cots_add_trip_chain_type_1(self):
+        """
+        Test for case 11 from the file "Enchainement Cas_API_20181010.xlsx"
+        1. A simple trip add with 5 stop_times all existing in navitia
```
Comment (mbounabi) [inline artemis/tests/guichet_unique_test.py]: Minor: A simple trip **addition**

File: artemis/tests/guichet_unique_test.py [hunk 8]
```diff
@@ -1052,6 +1052,281 @@ def test_kirin_cots_sequence_08(self):
...
+
+        """
+        All stop times in the trip are delayed by 15 minutes.
+        """
+        self.send_and_wait('trip_add_paris_marseille_with_delay_15.json')
+
+        """
+        Realtime request with same parameters as above
+        Should have a solution with departure at 12:15 and arrival at 17:15
+        """
```
Comment (mbounabi) [inline artemis/tests/guichet_unique_test.py]: Why not merge this comment with the one before 'send_and_wait' ?
(same for every other comments separated by a 'send_and_wait')

Comment (kadhikari) [reply artemis/tests/guichet_unique_test.py]: Because those are two different actions
